ballista/rust/scheduler/src/flight_sql.rs

```diff
@@ -52,22 +60,26 @@ use datafusion::prelude::SessionContext;
 use datafusion_proto::protobuf::LogicalPlanNode;
 use prost::Message;
 use tokio::time::sleep;
+use tonic::codegen::futures_core::Stream;
+use tonic::metadata::MetadataValue;
 use uuid::Uuid;
 
 pub struct FlightSqlServiceImpl {
     server: SchedulerServer<LogicalPlanNode, PhysicalPlanNode>,
     statements: Arc<Mutex<HashMap<Uuid, LogicalPlan>>>,
+    contexts: Arc<Mutex<HashMap<Uuid, Arc<SessionContext>>>>,
```

Comment:
Store a map of `SessionContext` to `UUID` so we can look up the session for the user and keep the tables registered.

TODO: move this into etcd?